openedx/core/djangoapps/programs/views.py

```diff
@@ -0,0 +1,68 @@
+"""
+Main views and method related to the Programs.
+"""
+
+import logging
+
+from openedx.core.djangoapps.util.helpers import get_id_token
+from openedx.core.djangoapps.programs.models import ProgramsApiConfig
+from openedx.core.djangoapps.programs.utils import programs_api_client
+
+
+log = logging.getLogger(__name__)
+# OAuth2 Client name for programs
+CLIENT_NAME = "programs"
+
+
+def get_course_programs_for_dashboard(user, course_keys):   # pylint: disable=invalid-name
+    """ Return all programs related to a user.
+
+    Given a user and an iterable of course keys, find all
+    the programs relevant to the user's dashboard and return them in a
+    dictionary keyed by the course_key.
+
+    Arguments:
+        user (user object): Currently logged-in User for which we need to get
+            JWT ID-Token
+        course_keys (list): List of course keys in which user is enrolled
+
+    Returns:
+        Dictionary response containing programs or None
+    """
+    course_programs = {}
+    if not ProgramsApiConfig.current().is_student_dashboard_enabled:
+        log.info("Programs service for student dashboard is disabled.")
```

Comment:
It's probably best to make this `log.warning` instead of `info` because in practice, this function should not even be getting called if not configured (there should be a check before calling it on the dashboard, which I see in #10372), but I like that we're being extra defensive here.